include/dynamatic/Dialect/Handshake/HandshakeInterfaces.td

```diff
@@ -57,13 +57,6 @@ def MergeLikeOpInterface : OpInterface<"MergeLikeOpInterface"> {
     if (!operands.size())
       return concreteOp.emitOpError("must have at least one data operand");
 
-    mlir::Type resultType = $_op->getResult(0).getType();
-
-    for (auto operand : operands)
-      if (operand.getType() != resultType)
-        return concreteOp.emitOpError("operand has type ") << operand.getType()
-            << ", but result has type " << resultType;
-
```

Comment:
This type checking between operands and result is unnecessary here.
For CMerge and Mux, the type is no longer the same between operands and result.
For MergeOp, `SameOperandsAndResultType` is already constrained